src/janitor.c

```diff
@@ -174,6 +174,12 @@ static void per_loop_activate(PgPool *pool)
 	PgSocket *client;
 	int sv_tested, sv_used;
 
+	/* if there is a cancelation request waiting, open a new socket */
```

Comment:
Maybe change this comment to

    if there is a cancel request waiting, open a new connection

Looks good otherwise.